sockeye/data_io.py

```diff
@@ -642,8 +676,12 @@ def get_prepared_data_iters(prepared_data_dir: str,
     for shard_fname in shard_fnames:
         check_condition(os.path.exists(shard_fname), "Shard %s does not exist." % shard_fname)
 
-    source_vocab_fname = os.path.join(prepared_data_dir, C.VOCAB_SRC_NAME) + C.JSON_SUFFIX
-    target_vocab_fname = os.path.join(prepared_data_dir, C.VOCAB_TRG_NAME) + C.JSON_SUFFIX
+    # MJP: Why was this hard-coded instead of reading from the data config?
```

Comment:
because these entries are None in the case where the vocab is created on the fly and not given on the commandline

Reply:
This means that the prepared data `data.config` file has "null" entries for the vocab, however. Is this intended? It seems more sensible for the `data.config` file to list the actual locations of the vocabularies, and then to load them from those locations, rather than hard-coding the path in the code and ignoring the value written to the config.

Here's a data config, for example:

```
shared_vocab: false
source: /redacted/path/train.bpe.lc.en.10
source_factor_vocabs:
- null
source_factors:
- /redacted/path/train.bpe.en.factor_case.10
target: /redacted/path/train.bpe.de.10
vocab_source: null
vocab_target: null
```